palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/PalisadeRestController.java

```diff
@@ -81,7 +77,7 @@ public PalisadeRestController(final PalisadeService palisadeService, final Error
             LOGGER.error("PalisadeRestController Exception thrown", e);
             httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
             Map<String, Object> attributes = new HashMap<>();
-            errorHandlingService.createErrorMessage(request, attributes, e);
+            palisadeService.errorMessage(request, token, attributes, e);
         }
         return ResponseEntity.status(httpStatus)
```

Comment:
palisadeResponse can be null when registerDataRequest is called and so token can also be null. These need to be handled in sending the errorMessage to the Audit plus the response back to the client.

Reply:
So the error is passed to the `error` topic and I don't think we need to passing the details of the internal errors to the client as the token has not been created.